documents/error.go

```diff
@@ -49,6 +55,26 @@ const (
 
 	// ErrDocumentProof must be used when document proof creation fails
 	ErrDocumentProof = errors.Error("document proof error")
+
+	// Document repository errors
+
+	// ErrDocumentRepositoryModelNotRegistered must be used when the model hasn't been registered in the database repository
+	ErrDocumentRepositoryModelNotRegistered = errors.Error("document model hasn't been registered in the database repository")
+
+	// ErrDocumentRepositorySerialisation must be used when document repository encounters a marshalling error
+	ErrDocumentRepositorySerialisation = errors.Error("document repository encountered a marshalling error")
+
+	// ErrDocumentRepositoryModelNotFound must be used when document repository can not locate the given model
+	ErrDocumentRepositoryModelNotFound = errors.Error("document repository could not locate the given model")
+
+	// ErrDocumentRepositoryModelSave must be used when document repository can not save the given model
+	ErrDocumentRepositoryModelSave = errors.Error("document repository could not save the given model")
+
+	// ErrDocumentRepositoryModelAllReadyExists must be used when document repository finds an already existing model when saving
+	ErrDocumentRepositoryModelAllReadyExists = errors.Error("document repository found an already existing model when saving")
```

Comment:
`ErrDocumentRepositoryModelAllReadyExists` => `ErrRepositoryModelExists `

Reply:
`document repository found an already existing model when saving` => `model already exists in the repository`